apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLogger.cs

```diff
@@ -0,0 +1,201 @@
+        /// <param name="version">Represents the source code version that the developer
+        ///     provided. Cannot be null.</param> 
+        public static ErrorReportingExceptionLogger Create(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version) =>
+            Create(Task.FromResult(client), projectId, serviceName, version);
+
+        /// <summary>
+        /// Creates an instance of <see cref="ErrorReportingExceptionLogger"/>
+        /// </summary>
+        /// <param name="clientTask">The Error Reporting client. Cannot be null.</param>
+        /// <param name="projectId">The Google Cloud Platform project ID. Cannot be null.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the 
+        ///     executable or job. Cannot be null.</param>
+        /// <param name="version">Represents the source code version that the developer
+        ///     provided. Cannot be null.</param> 
+        public static ErrorReportingExceptionLogger Create(
+            Task<ReportErrorsServiceClient> clientTask, string projectId, string serviceName, string version) =>
+            new ErrorReportingExceptionLogger(clientTask, projectId, serviceName, version);
+
+        /// <summary>
+        /// Creates an instance of <see cref="ErrorReportingExceptionLogger"/> using credentials as
+        /// defined by <see cref="GoogleCredential.GetApplicationDefaultAsync"/>.
+        /// </summary>
+        /// <param name="projectId">The Google Cloud Platform project ID. Cannot be null.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the 
+        ///     executable or job. Cannot be null.</param>
+        /// <param name="version">Represents the source code version that the developer
+        ///     provided. Cannot be null.</param> 
+        public static ErrorReportingExceptionLogger Create(
+            string projectId, string serviceName, string version) =>
+            Create(ReportErrorsServiceClient.CreateAsync(), projectId, serviceName, version);
+
+        private ErrorReportingExceptionLogger(
+           Task<ReportErrorsServiceClient> clientTask, string projectId, string serviceName, string version)
+        {
+            GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
+            _clientTask = GaxPreconditions.CheckNotNull(clientTask, nameof(clientTask));
+            _projectName = new ProjectName(projectId);
+            _serviceContext = new ServiceContext
+            {
+                Service = GaxPreconditions.CheckNotNull(serviceName, nameof(serviceName)),
+                Version = GaxPreconditions.CheckNotNull(version, nameof(version)),
+            };
+        }
+
+        /// <summary>
+        /// Asynchronously reports an exception that occurred to the Stackdriver Error Reporting API.
+        /// </summary>
+        /// <returns>An empty response on success.</returns>
```

Comment:
This should be something like "A task containing an empty..." or "A task representing the asynchronous operation. When complete, the result is an empty..."